SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java

```diff
@@ -148,8 +155,17 @@ private boolean writeTailMetadata(boolean finished) {
     return jsonObjectFileHelper.writeObject(tailMetadata, path, log);
   }
 
+  /**
+   * Return a String for generating a PathMatcher.
+   * The matching files are caught by the S3 Uploader and pushed to S3.
+   * @return file glob String.
+   */
   private String getS3Glob() {
-    return String.format("%s*.gz*", taskDefinition.getServiceLogOutPath().getFileName());
+    List<String> fileNames = new LinkedList<>();
+    fileNames.add(taskDefinition.getServiceLogOutPath().getFileName().toString());
+    fileNames.addAll(Arrays.asList(configuration.getS3FilesToBackup()));
+
+    return String.format("{%s}*.gz*", StringUtils.join(fileNames, ","));
```

Comment:
is this valid glob syntax? also for joinining, use guava's `Joiner.on(",").join()` instead of `StringUtils.join()` (not because StringUtils is necessarily bad, but it's best to standardize on the guava library here)